docs/development/component-checklist.md

```diff
@@ -24,11 +24,12 @@ This document provides a checklist for them that you can walk through.
    We define all image references centrally in the [`charts/images.yaml`](https://github.com/gardener/gardener/blob/6a0fea86850ffec8937d1956bdf1a8ca6d074f3b/charts/images.yaml) file.
    Hence, the image references must not be hard-coded in the pod template spec but read from this so-called [image vector](../deployment/image_vector.md) instead.
 
-4. **Do not use `docker.io` container registry**
+4. **Do not use container images from Docker Hub** (example: [image vector](https://github.com/gardener/gardener/blob/6f4e64fe9494cafb5c5da9a2c0a491a5690161b6/charts/images.yaml#L257-L260), [prow configuration](https://github.com/gardener/ci-infra/blob/92782bedd92815639abf4dc14b2c484f77c6e57d/config/images/images.yaml#L7-L10))
 
    The `docker.io` registry doesn't support pulling images over IPv6 (see [Beta IPv6 Support on Docker Hub Registry](https://www.docker.com/blog/beta-ipv6-support-on-docker-hub-registry/)).
    There is also a strict [rate-limit](https://docs.docker.com/docker-hub/download-rate-limit/) that applies to the Docker Hub registry.
-   If you need an image from `docker.io`, please use the [Google Mirror](https://cloud.google.com/container-registry/docs/pulling-cached-images) (`mirror.gcr.io`) instead to circumvent these issues.
+   There is a [prow job](https://github.com/gardener/ci-infra/blob/92782bedd92815639abf4dc14b2c484f77c6e57d/config/jobs/ci-infra/copy-images.yaml) copying images from Docker Hub that are needed in gardener components to the gardener GCR under the prefix `eu.gcr.io/gardener-project/3rd/` (see the [documentation](https://github.com/gardener/ci-infra/tree/master/config/images) or [gardener/ci-infra#619](https://github.com/gardener/ci-infra/issues/619)).
+   If you want to use a new image from Docker Hub or upgrade an already used image to a newer tag, please open a PR to the ci-infra repository that modifies the job's list of images to copy: [`images.yaml`](https://github.com/gardener/ci-infra/blob/master/config/images/images.yaml).
```

Comment:
I guess the PR to ci-infra needs to be merged before you can update the corresponding image in gardener/gardener. Is my understanding correct?

Would it make sense to have some automation in place or do you envision this double maintenance as standard going forward?

Reply:
Yes, that is correct. 

> Would it make sense to have some automation in place or do you envision this double maintenance as standard going forward?

I think this "double maintenance" is the simplest option we have that fulfills our requirements.
I tried to explain the reasoning in https://github.com/gardener/ci-infra/issues/619:

> We don't want to add repository-specific jobs based on their image vectors (`images.yaml`) because
> 
> * images specified in the image vector need to be present in the destination registry for pull request jobs to succeed
>   
>   * however, a CI job should only copy the images once a PR got approved and merged
>   * this introduces a chicken-egg problem for adding new features based on new images
>   * having an out-of-band configuration (centralized in this repository) allows configuring the copying first and then adding the new feature based on that
> * only a subset of gardener's repositories use prow (some are still using concourse): configuring decentralized jobs everywhere in different tools is a high effort and troublesome in maintenance

Do you agree with the reasoning? 
I'm also open to better ideas if you have any :)

Reply:
I understand the reasoning why it cannot start from the PR in the repository actually using the image.
However, what do you think about something like https://github.com/gardener/gardener/pull/7685 for this, i.e. updating the image in ci-infra and a bot automatically opening update pull requests in the affected repositories? That would reduce the double maintenance to single maintenance in ci-infra + approving of PRs in the real repositories.
WDYT?

Reply:
Ok, got the idea.
However, I think this wouldn't fit all cases. E.g., where an image update needs to happen together with code changes (incompatible flag changes, etc.). In such cases, you would be back to opening/managing two PRs.
I'm skeptical that the additional effort would be worth the benefits, given that updating a few lines of YAML is really straight-forward.
Hence, I suggest going forward with the current mechanism and adding more automation if the burden grows too large.

Reply:
Changing the few lines is certainly not a big effort, but keeping in mind that you need to change another repository first adds to the overall complexity. I am not a big fan of that.
Anyway, we can start with this approach add improve when the annoyance gets to big.